Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/edit/EditCheckInViewModel.kt

```diff
@@ -0,0 +1,160 @@
+
+    fun onClose() {
+        events.value = EditCheckInNavigation.BackNavigation
+    }
+
+    fun onStartDateClicked() {
+        val utcDateTime = checkInStartTime.value?.toDateTime()
+        openStartPickerEvent.value = DateTimePickerEvent.DatePickerEvent(utcDateTime?.toLocalDate())
+    }
+
+    fun onStartTimeClicked() {
+        val utcDateTime = checkInStartTime.value?.toDateTime()
+        openStartPickerEvent.value = DateTimePickerEvent.TimePickerEvent(utcDateTime?.toLocalTime())
+    }
+
+    fun onEndDateClicked() {
+        val utcDateTime = checkInEndTime.value?.toDateTime()
+        openEndPickerEvent.value = DateTimePickerEvent.DatePickerEvent(utcDateTime?.toLocalDate())
+    }
+
+    fun onEndTimeClicked() {
+        val utcDateTime = checkInEndTime.value?.toDateTime()
+        openEndPickerEvent.value = DateTimePickerEvent.TimePickerEvent(utcDateTime?.toLocalTime())
+    }
+
+    fun onStartTimeChanged(event: DateTimePickerEvent) {
+        val startDateTime = checkInStartTime.value?.toDateTime()
+
+        when (event) {
+            is DateTimePickerEvent.TimePickerEvent ->
+                checkInStartTime.value = startDateTime?.withTime(event.localTime)?.toInstant()
+            is DateTimePickerEvent.DatePickerEvent ->
+                checkInStartTime.value = startDateTime?.withDate(event.localDate)?.toInstant()
+        }
+    }
+
+    fun onEndTimeChanged(event: DateTimePickerEvent) {
+        val endDateTime = checkInEndTime.value?.toDateTime()
+
+        when (event) {
+            is DateTimePickerEvent.TimePickerEvent ->
+                checkInEndTime.value = endDateTime?.withTime(event.localTime)?.toInstant()
+            is DateTimePickerEvent.DatePickerEvent ->
+                checkInEndTime.value = endDateTime?.withDate(event.localDate)?.toInstant()
+        }
+    }
+
+    fun onSaveClicked() {
+        launch {
+            val oldCheckIn = checkInFlow.value
+            val newCheckInTime = checkInStartTime.value
+            val newCheckOutTime = checkInEndTime.value
+
+            if (oldCheckIn != null && newCheckInTime != null && newCheckOutTime != null) {
+                checkInRepository.updateCheckIn(checkInId = oldCheckIn.id) {
+                    it.copy(
+                        checkInStart = newCheckInTime,
+                        checkInEnd = newCheckOutTime
+                    )
+                }
+            }
+            events.postValue(EditCheckInNavigation.ConfirmNavigation)
+        }
+    }
+
+    @AssistedFactory
+    interface Factory : CWAViewModelFactory<EditCheckInViewModel> {
+        fun create(
+            editCheckInId: Long?
+        ): EditCheckInViewModel
+    }
+
+    data class UiState(
+        private val checkIn: CheckIn,
+        private val checkInStartInstant: Instant,
+        private val checkInEndInstant: Instant
+    ) {
+        val description: String get() = checkIn.description
+        val typeRes: Int get() = mapTraceLocationToTitleRes(checkIn.type)
```

Comment:
`@StringRes` ?

Reply:
i cannot annotate a getter afaik

Reply:
`@get:StringRes ` not a blocker for me

Reply:
ah right, completely forgot about that :D